yt_dlp/extractor/tencent.py

```diff
@@ -63,27 +78,45 @@ def _get_video_api_response(self, video_url, video_id, series_id, subtitle_forma
 
     def _extract_video_formats_and_subtitles(self, api_response, video_id):
         video_response = api_response['vl']['vi'][0]
-        video_width, video_height = video_response.get('vw'), video_response.get('vh')
 
         formats, subtitles = [], {}
         for video_format in video_response['ul']['ui']:
             if video_format.get('hls') or determine_ext(video_format['url']) == 'm3u8':
                 fmts, subs = self._extract_m3u8_formats_and_subtitles(
                     video_format['url'] + traverse_obj(video_format, ('hls', 'pt'), default=''),
                     video_id, 'mp4', fatal=False)
-                for f in fmts:
-                    f.update({'width': video_width, 'height': video_height})
 
                 formats.extend(fmts)
                 self._merge_subtitles(subs, target=subtitles)
             else:
                 formats.append({
                     'url': f'{video_format["url"]}{video_response["fn"]}?vkey={video_response["fvkey"]}',
-                    'width': video_width,
-                    'height': video_height,
                     'ext': 'mp4',
                 })
 
+        identifier = video_response.get('br')
+        format_response = traverse_obj(
+            api_response, ('fl', 'fi', lambda _, v: v['br'] == identifier),
+            expected_type=dict, get_all=False) or {}
```

Comment:
Avoid variables only used once

Reply:
This is not used only once. `identifier` is cached bc otherwise we would call `video_response.get('br')` each time for each item in the `traverse_obj`/

Reply:
I agree with what @Grub4K  said, it's worth avoiding repeated calls in loops.

Reply:
It's just a dict access, and doesn't really matter. But I wont force it